app/services/katello/applicability/applicable_content_helper.rb

```diff
@@ -0,0 +1,88 @@
+module Katello
+  module Applicability
+    class ApplicableContentHelper
+      attr_accessor :content_facet, :content_unit_class
+
+      def initialize(content_unit_class, content_facet = nil)
+        self.content_facet = content_facet
+        self.content_unit_class = content_unit_class
+      end
+
+      def calculate_and_import
+        to_add, to_remove = applicable_differences
+        ActiveRecord::Base.transaction do
+          insert(to_add) unless to_add.blank?
+          remove(to_remove) unless to_remove.blank?
+        end
+
+        # FIXME: Debug return
+        return [to_add, to_remove]
+      end
+
+      def content_ids
+        if self.content_unit_class == ::Katello::Erratum
+          # Query for all Errata ids that are attached to the host's applicable packages
+          return ::Katello::Erratum.joins("INNER JOIN katello_repository_errata ON katello_errata.id = \
+                                          katello_repository_errata.erratum_id").
+                                          where(id: ::Katello::ErratumPackage.
+                                                where(nvrea: self.content_facet.applicable_rpms.pluck(:nvra)).
+                                                pluck(:erratum_id)).pluck(:id).uniq
```

Comment:
I think we have to use a similar mechanism as below to limit the errata to what is in the host's bound repositories (or the library instances of them)